python/ray/serve/_private/router.py

```diff
@@ -495,6 +495,25 @@ async def _resolve_deployment_responses(
         # Return new args and new kwargs
         return new_args, new_kwargs
 
+    def process_finished_request(self, replica_id: ReplicaID, result):
+        self._metrics_manager.dec_num_running_requests_for_replica(replica_id)
+        if isinstance(result, ActorDiedError):
+            # Replica has died but controller hasn't notified the router yet.
+            # Don't consider this replica for requests in the future, and retry
+            # scheduling request.
+            self._replica_scheduler.on_replica_actor_died(replica_id)
+            logger.warning(
+                f"{replica_id} will not be considered for future "
+                "requests because it has died."
+            )
```

Comment:
hm probably this message should just be logging inside the callback so it's consistent across callsites

Reply:
yes and also during active probing

Reply:
@edoakes Hmm, took a look at code and might be hard to unify. For active probing, tasks are launched + processed in the scheduler, and it seems more straightforward to directly deal with exceptions from probing tasks instead of using a ray object ref callback.